### DISCUSSION TITLE
Bad Allocation in phase 3 - Help

### DISCUSSION THREAD

**Discussion by fabiomesmu** [swar/Swar-Chia-Plot-Manager#1047] [Q&A]
Good morning everyone, I'm losing a lot of plots when they start phase 3, I made several adjustments but I couldn't solve them, I create 1 plot and I lose 3;
Here's the configuration and the error, if anyone can help...

Ryzen5 - 6/12
32gb Ram
2tb nvme

Config

name: XHA_007
    max_plots: 999
    farmer_public_key:
    pool_public_key:
    temporary_directory: R: \
    temporary2_directory:
    destination_directory: G: \
    size: 32
    bitfield: true
    threads: 12
    buckets: 128
    memory_buffer: 30510
    max_concurrent: 8
    max_concurrent_with_start_early: 9
    initial_delay_minutes: 0
    stagger_minutes: 5
    max_for_phase_1: 3
    concurrency_start_early_phase: 4
    concurrency_start_early_phase_delay: 0
    temporary2_destination_sync: false
    exclude_final_directory: false
    skip_full_destinations: true
    unix_process_priority: 10
    windows_process_priority: 32
    enable_cpu_affinity: false
    cpu_affinity: [0, 1, 2, 3, 4, 5]

Error:

Starting phase 3/4: Compression from tmp files into "R: \\ plot-k32-2021-05-30-17-34-b3226e5bfac64db842c57352217290f036a3a77b82c441010caa5a4ad0e2cc4b.plot.2.tmp" ... Sun May 30 23:23:27 2021
Compressing tables 1 and 2
Caught plotting error: bad allocation
[7360] Failed to execute chia script
Traceback (most recent call last):
  File "chia \ cmds \ chia.py", line 81, in <module>
  File "chia \ cmds \ chia.py", line 77, in main
  File "click\core.py", line 829, in __call__
  File "click\core.py", line 782, in main
  File "click \ core.py", line 1259, in invoke
  File "click \ core.py", line 1259, in invoke
  File "click \ core.py", line 1066, in invoke
  File "click\core.py", line 610, in invoke
  File "click \ decorators.py", line 21, in new_func
  File "chia \ cmds \ plots.py", line 135, in create_cmd
  File "chia \ plotting \ create_plots.py", line 176, in create_plots
RuntimeError: bad allocation

Thank You

**Comment by vortex100cc** [comment]
"RuntimeError: bad allocation" - this error happens when you have issues with your RAM. 
"
threads: 12
buckets: 128
memory_buffer: 30510
max_concurrent: 8
"
First of all, you need to assign threads and mem_buffer amount for just one plot in this job, not like for all 8 plots you're trying to run concurrently. In your case plot manager trying to use almost all the memory you have just for one plot. Though actually 4000-5000 mem buffer is more than enough.
Second, most likely you woun't be able to run 8 plots in a row with your setup. 2tb of temp drive for sure can handle 6 plots, maybe 7 with one in start early phase.
So, in your case I can suggest to test this :
"
threads: 2
buckets: 128
memory_buffer: 5000
max_concurrent: 6
max_concurrent_with_start_early: 7
"

And one more thing. In my case bad allocation error happaned because of small pagefile. I Just added a system designeted pagefile to two more drives and that solved the problem. Also it can happen if you run something resourse intensive on your pc while plotting, but it's occasionally, for xmpl mining other stuff like ETH. 
So just fix your config and add more pagefile and you'll be good
